### ISSUE TITLE
Ludo crashes when trying to scan GBA ROMs folder

### ISSUE THREAD

**Issue by LOuroboros** [libretro/ludo#463]
Basically what I say in the title.
I have my collection of GBA ROMs located in "E:\Lunos\Emulation\ROMs\GB, GBC and GBA", and when I go there by navigating through folders, Ludo closes right away.

https://user-images.githubusercontent.com/4485172/130670756-0e674352-7faa-4f0d-b767-9cd287716e73.mp4

**Comment by kivutar** [comment]
OK thanks for the bug report.

Can I get a list of the content of this folder?

Do you know how to launch Ludo from the command line to get a log or a stack trace?

**Comment by LOuroboros** [comment]
> Can I get a list of the content of this folder?

The folder contains multiple folders. In each folder there's a ROM, and in some cases there's a couple of extra files. Documentation, savefile or savestate files generated by either VBA or mGBA and such.

> Do you know how to launch Ludo from the command line to get a log or a stack trace?

Well, "opening from the command line" sounds as easy as opening Ludo.exe in the Windows Command Prompt, no?

But I've no idea about how to generate a log or a stack trace :eyes:

**Comment by kivutar** [comment]
Yes opening ludo.exe  from the command prompt should be enough, and you can pass a flag like `-v` to get more logs.

**Comment by LOuroboros** [comment]
> 
> 
> Yes opening ludo.exe from the command prompt should be enough, and you can pass a flag like `-v` to get more logs.

I don't think it's working?

Booting up Ludo by using `ludo.exe -v` doesn't seem to record anything on the terminal, and if I hit "Scan this directory" the program closes just like it happened a few weeks back when I opened this issue post :/ ...

![Untitled](https://user-images.githubusercontent.com/4485172/132627036-4e5f79a0-3770-4c79-994b-67420077128a.png)

**Comment by kivutar** [comment]
Maybe on Windows passing a flag would be done with another syntax. Try `ludo.exe /v` maybe?

**Comment by LOuroboros** [comment]
> 
> 
> Maybe on Windows passing a flag would be done with another syntax. Try `ludo.exe /v` maybe?

Still nothing :/

![explorer_894](https://user-images.githubusercontent.com/4485172/132738436-b4988d6e-15bd-4fcd-aca1-0b166ec32455.png)

**Comment by kivutar** [comment]
Ah sorry for that... I don't know how to get enough information to fix this.

Maybe you can try creating a new folder where you would transfer the GB roms one by one to see if there is one in particular that causes the crash?

**Comment by LOuroboros** [comment]
> 
> 
> Ah sorry for that... I don't know how to get enough information to fix this.
> 
> Maybe you can try creating a new folder where you would transfer the GB roms one by one to see if there is one in particular that causes the crash?

Something tells me the crash is being caused by the fact that I'm trying to make Ludo read a folder with ROMs allocated in an external hard drive *(`E:`)* and the program just doesn't like that, but I'll try to move the folder to my desktop and scan it from there anyway. Will edit in a moment.

**Comment by kivutar** [comment]
ROM hacks not being scanned is a separate issue, due to the scanner being checksum based.

What filesystem was this E drive?

**Comment by LOuroboros** [comment]
> 
> 
> ROM hacks not being scanned is a separate issue, due to the scanner being checksum based.
> 
> What filesystem was this E drive?

It's NTFS.

**Comment by tompruitt** [comment]
@LOuroboros can you still reproduce it?

**Comment by LOuroboros** [comment]
> @LOuroboros can you still reproduce it?

Yup. Ludo v0.17.2 crashes the moment I hit the X button to scan my `D:\Lunos\Emulation\ROMs\GB, GBC and GBA` directory, @tompruitt.

*I have the external HDD where I keep my ROMs and emulators on `D:` now. Windows things. I formatted my PC about 2 times during the past 3 years and during my last formatting Windows just decided to host it on `D:` instead of `E:`.*

**Comment by LOuroboros** [comment]
For whatever it's worth, I checked something at random which was to rename my "GB, GBC and GBA" folder to "GBGBCGBA", but Ludo still crashed regardless.

I reverted the folder name change, then copy-pasted the folder in my Windows desktop *(`C:\Users\Lunos\Desktop`)* and that did work just fine.

![%pn_20240330_132401153](https://githcom/libretro/ludo/assets/4485172/89fa648e-25ad-4802-9b5f-ccd7b27ae13d)

![%pn_20240330_132717578](https://github.com/libretro/ludo/assets/4485172/b327ee2f-e1b2-41f1-9f3f-5bb76584abcc)

It's a shame that Ludo still can't recognize ROM Hacks despite being completely valid and functional GBA ROM files though.

*Also, ignore the black bars on the sides in the pictures above. I had to use an external application called Magpie in order to set Ludo to Borderless Fullscreen because the program doesn't seem to offer an option for that in its settings 😅 ...*
